src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -582,6 +608,11 @@ protected Area awtArea()
         return this.awtArea;
     }
 
+    private boolean awtOverflows()
+    {
+        return this.bounds().width().asDm7() < 0 || this.bounds().height().asDm7() < 0;
```

Comment:
this looks like the ```Rectangle``` class overflows when the bounds are too large.  Could the dm7 value in Rectangle be a long instead of an int?  Then the checks here could be a specific > MAX_INT check.